bridgesync/downloader.go

```diff
@@ -115,8 +120,225 @@ func buildAppender(
 	return appender, nil
 }
 
+// Transaction represents the structure of a transaction returned by eth_getTransactionByHash
+type Transaction struct {
+	FromRaw          string `json:"from"`
+	To               string `json:"to"`
+	Hash             string `json:"hash"`
+	Value            string `json:"value"`
+	Gas              string `json:"gas"`
+	GasPrice         string `json:"gasPrice"`
+	Nonce            string `json:"nonce"`
+	Input            string `json:"input"`
+	BlockHash        string `json:"blockHash"`
+	BlockNumber      string `json:"blockNumber"`
+	TransactionIndex string `json:"transactionIndex"`
+}
+
+func (t *Transaction) From() common.Address {
+	return common.HexToAddress(t.FromRaw)
+}
+
+func RPCTransactionByHash(client aggkittypes.EthClienter,
+	txHash common.Hash) (*Transaction, error) {
+	// Use client.Call to fetch transaction details using eth_getTransactionByHash
+	var tx Transaction
+	err := client.Call(&tx, "eth_getTransactionByHash", txHash.Hex())
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch transaction by hash: %w", err)
+	}
+	return &tx, nil
+}
```

Comment:
This function is not needed, you can invoke, from `go-ethereum`. The `aggkittypes.EthClienter` already implements the `TransactionReader`:

```go
type TransactionReader interface {
   TransactionByHash(ctx context.Context, txHash common.Hash) (tx *types.Transaction, isPending bool, err error)
   ...
}
```
```suggestion
```

Also as @vcastellm pointed out, we can then remove the custom `Transaction` struct and reuse the one from go-ethereum.